manifests/install_systemd_unit.pp

```diff
@@ -2,15 +2,49 @@
 	$unit_options = {},				# Arguments to pass to the systemd [Unit] section of template
 	$service_options = {},			# Arguments to pass to the systemd [Service] section of template
 	$install_options = {},			# Arguments to pass to the systemd [Install] section of template
+	$timer_options = {},			# Arguments to pass to the systemd [Timer] section of the timer template
 	$service_ensure = "running",	# State of the service after installed
-	$service_enable = true			# Enabled state of the service after installed
+	$service_enable = true,			# Enabled state of the service after installed
+	$path = "/etc/systemd/system",	# Path to systemd files
 ) {
 
-	::systemd::unit_file { "${name}.service":
-		content => template("${module_name}/systemd_service.erb")
-	} -> service { $name:
-		ensure => $service_ensure,
-		enable => $service_enable,
-		require => Exec['systemctl-daemon-reload'],
+	if $name =~ /^([A-Za-z0-9_-]+)(\.)(timer)$/ {
+		$real_name = $name
+		$timer_service = regsubst($real_name, '.timer', '.service', 'G')
+		$service = ""
+		$exec = "systemctl start ${real_name}"
+	} elsif $name =~ /^([A-Za-z0-9_-]+)(\.)(service)$/ {
+		$real_name = $name
+		$service = regsubst($real_name, '.service', '', 'G')
+		$exec = ""
+	} else {
+		$real_name = "${name}.service"
+		$service = regsubst($real_name, '.service', '', 'G')
+		$exec = ""
+		notice("${name} does not include '.service' or '.timer', assuming '.service'")
 	}
-}
+
+	::systemd::unit_file { "${real_name}":
+		content => template("${module_name}/systemd_unit.erb"),
+		path => $path,
+	}
+
+	if !empty($service) {
+		service { $service:
+			ensure => $service_ensure,
+			enable => $service_enable,
+			notify => Exec['systemctl-daemon-reload'],
```

Comment:
I think I remember now why this was needed as a `require`, notify won't help matters because daemon-reload reloads configuration, so if the service needs to be triggered, notifying won't do anything. Then again we've been running servicetools without the require for a long while now and no issues. Probably best to just ditch the notify here altogether.